File: Solution/WebSolution/Dotnet/AssessmentPortal/Monolithic/Monolithic_Controller/AssessmentServices/Implementations/ResultService.cs
using Transflower.TFLAssessment.Entities;
using Transflower.TFLAssessment.Services.Interfaces;
using Transflower.TFLAssessment.Repositories.Interfaces;
namespace  Transflower.TFLAssessment.Services;

public class ResultService : IResultService
{
    private readonly IResultRepository _repository;

    public ResultService()
    {

    }
    public ResultService(IResultRepository repository)
    {
        _repository = repository;
    }

    public async Task<int> GetCandidateScore(int candidateId, int testId)
    {
        return await _repository.GetCandidateScore(candidateId, testId);
    }

    public async Task<bool> SetCandidateTestStartTime(int candidateId, int testId, TestTime time)
    {
        return await _repository.SetCandidateTestStartTime(candidateId, testId, time);
    }

    public async Task<bool> SetCandidateTestEndTime(int candidateId, int testId, TestTime time)
    {
        return await _repository.SetCandidateTestEndTime(candidateId, testId, time);
    }

    public async Task<CandidateResultDetails> CandidateTestResultDetails(int candidateId, int testId)
    {
        return await _repository.CandidateTestResultDetails(candidateId, testId);
    }

    public async Task<List<TestResultDetails>> GetTestResultDetail(int testId)
    {
        return await _repository.GetTestResultDetail(testId);
    }

    public async Task<List<AppearedCandidate>> GetAppearedCandidates(int testId)
    {
        return await _repository.GetAppearedCandidates(testId);
    }

    public async Task<List<PassedCandidateDetails>> GetPassedCandidateResults(int testId)
    {
        return await _repository.GetPassedCandidateResults(testId);
    }

    public async Task<List<FailedCandidateDetails>> GetFailedCandidateResults(int testId)
    {
        return await _repository.GetFailedCandidateResults(testId);
    }

    public async Task<bool> SetPassingLevel(int testId, int passingLevel)
    {
        return await _repository.SetPassingLevel(testId, passingLevel);
    }

    public async Task<List<CandidateSubjectResults>> GetSubjectResultDetails(int subjectId)
    {
        return await _repository.GetSubjectResultDetails(subjectId);
    }

    public async Task<List<TestList>> GetTestList(int candidateId)
    {
        return await _repository.GetTestList(candidateId);
    }

    public async Task<List<Subject>> GeAllSubjects()
    {
        return await _repository.GeAllSubjects();
    }

    public async Task<List<TestAverageReport>> GetTestAverageReport(int testId)
    {
        return await _repository.GetTestAverageReport(testId);
    }
    public async Task<List<TestScoreDto>> GetCandidateAllScore(int candidateId)
    {
        return await _repository.GetCandidateAllScore(candidateId);
    }
}

File: Solution/WebSolution/Java/AssessmentAPI/src/main/java/com/transflower/tflassessment/services/ResultService.java
package com.transflower.tflassessment.services;

import java.time.LocalDateTime;
import java.util.List;
import java.util.concurrent.CompletableFuture;

import com.transflower.tflassessment.entities.AppearedCandidate;
import com.transflower.tflassessment.entities.CandidateResultDetails;
import com.transflower.tflassessment.entities.CandidateSubjectResults;
import com.transflower.tflassessment.entities.FailedCandidateDetails;
import com.transflower.tflassessment.entities.PassedCandidateDetails;
import com.transflower.tflassessment.entities.Subject;
import com.transflower.tflassessment.entities.TestAverageReport;
import com.transflower.tflassessment.entities.TestList;
import com.transflower.tflassessment.entities.TestResultDetails;
import com.transflower.tflassessment.entities.TestScoreDto;

public interface ResultService {

    public CompletableFuture<Integer> getCandidateScore(int candidateId, int testId);
    // public Task<int> GetCandidateTestScore(int candidateId, int testId);
    public CompletableFuture<Boolean> setCandidatTeststarttime(int candidateId, int testId, LocalDateTime time) ;

    public CompletableFuture<Boolean> setCandidateTestEndTime(int candidateId, int testId, LocalDateTime time);

    public CompletableFuture<CandidateResultDetails> CandidateTestResultDetails(int candidateId, int testId);

    public CompletableFuture <List<AppearedCandidate>> getAppearedCandidates(int testId);

    public CompletableFuture<List<TestList>> getTestList(int candidateId);

    public CompletableFuture <List<PassedCandidateDetails>> getPassedCandidateResults(int testId);

    public CompletableFuture<List<FailedCandidateDetails>> getFailedCandidateResults(int testId);

    public CompletableFuture<Boolean> setPassingLevel (int testId, int passingLevel);

    public CompletableFuture<List<CandidateSubjectResults>> getSubjectResultDetails(int subjectId);

    // public Task<int[]> GetAllTestIds();
    public CompletableFuture<List<Subject>> getAllSubjects();

    public  CompletableFuture<List<TestAverageReport>> getTestAverageReport(int testId);

    public CompletableFuture<List<TestScoreDto>> getCandidateAllScore(int candidateId);
         
    public CompletableFuture<List<TestResultDetails>> getTestResultDetail(int testId);

    public CompletableFuture<Boolean> setCandidateTestResultDetails(int candidateId, int testId);



}

